src/order/well_founded_set.lean

```diff
@@ -87,24 +87,45 @@ instance is_strict_order.subset {s : set α} {r : α → α → Prop} [is_strict
   to_is_trans := ⟨λ a b c ab bc, ⟨trans_of r ab.1 bc.1, ab.2.1, bc.2.2⟩ ⟩ }
 
 theorem well_founded_on_iff_no_descending_seq {s : set α} {r : α → α → Prop} [is_strict_order α r] :
-  s.well_founded_on r ↔ ∀ (f : ((>) : ℕ → ℕ → Prop) ↪r r), ¬ (range f) ⊆ s :=
+  s.well_founded_on r ↔ ∀ (f : ((>) : ℕ → ℕ → Prop) ↪r r), ¬∀ n, f n ∈ s :=
```

Comment:
Any reason you're not using this?
```suggestion
  s.well_founded_on r ↔ ∀ (f : ((>) : ℕ → ℕ → Prop) ↪r r), ∃ n, f n ∉ s :=
```

Reply:
I started with this but then I found myself rewriting `<- not_forall` all the time.